samples/create_job.py

```diff
@@ -13,33 +13,46 @@
 # limitations under the License.
 
 import typing
+from typing import Union
 
 if typing.TYPE_CHECKING:
-    from google.cloud import bigquery
+    from google.cloud.bigquery import LoadJob, CopyJob, ExtractJob, QueryJob
 
 
-def create_job() -> "bigquery.QueryJob":
+def create_job() -> "Union[LoadJob, CopyJob, ExtractJob, QueryJob]":
 
     # [START bigquery_create_job]
     from google.cloud import bigquery
 
     # Construct a BigQuery client object.
     client = bigquery.Client()
 
-    query_job = client.query(
-        "SELECT country_name from `bigquery-public-data.utility_us.country_code_iso`",
-        # Explicitly force job execution to be routed to a specific processing
-        # location.
-        location="US",
-        # Specify a job configuration to set optional job resource properties.
-        job_config=bigquery.QueryJobConfig(
-            labels={"example-label": "example-value"}, maximum_bytes_billed=1000000
-        ),
-        # The client libraries automatically generate a job ID. Override the
-        # generated ID with either the job_id_prefix or job_id parameters.
-        job_id_prefix="code_sample_",
+    query_job = client.create_job(
+        # Specify a job configuration, providing both query
+        # and to set optional job resource properties, if needed.
+        # The job instance can be a LoadJob, CopyJob, ExtractJob, QueryJob
+        # Here, we demonstrate a "query" job.
+        # Reference: https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.create_job
+        #
+        # NOTE: the preferred approach is to use one of the dedicated API calls:
```

Comment:
Could you elaborate on what "preferred approach" should be for?

If it's for executing the query it seems counterproductive to have a sample for `create_client` if this is not the preferred method.

Reply:
@tswast can you elaborate for us on this comment you made in the issue, as it relates to @dandhlee's question above?

> "That section is about any kind of job, not just queries. As such, it should use the [create_job](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.create_job) method instead of the more specific query method. There should be comments that it is recommended to use the corresponding method for query/copy/load/extract."

Reply:
One use-case this sample would be preferred to compared to one of the more specific examples is when folks try to retry failed jobs.

There are customers who use this method by iterating through a list of recent jobs and retrying any that have failed as a way to make their data pipelines a bit more robust.

One could also use this method as a way to create a job with an experimental API property that hasn't been added to the client library's manually written job configuration classes yet.